packages/block-library/src/block/index.js

```diff
@@ -20,3 +21,29 @@ export const settings = {
 };
 
 export const init = () => initBlock( { name, metadata, settings } );
+
+// In order to make sure that blocks retain their full/wide alignments in the editor when converted to synced patterns
+// we need to add the layout support to the core/block block wrapper. This is only needed in the editor as there is
+// no additional wrapper in the front-end so adding here instead of using the usual supports mechanism in block.json.
+function addEditorOnlyLayoutSupportForPatterns( blockSettings ) {
+	if ( blockSettings.name === 'core/block' ) {
+		if ( ! blockSettings.supports ) {
+			blockSettings.supports = {};
+		}
+		blockSettings.supports.layout = {
+			allowSizingOnChildren: true,
+			allowEditing: false,
+			default: {
+				type: 'constrained',
+			},
+		};
+	}
+
+	return blockSettings;
+}
+
+addFilter(
+	'blocks.registerBlockType',
+	'core/synced-patterns/add-editor-only-layout-support-for-patterns',
+	addEditorOnlyLayoutSupportForPatterns
+);
```

Comment:
Why is this done in a filter instead of just updating the block.json?

Reply:
Because we don't want the layout supports serialized in the frontend by the server-side rendering. Without the wrapper on the frontend it adds the layout classes to the first child block instead which causes issues.

We did look at adding `__experimentalSkipSerialization` to layouts, but it was a bit quirky only applying this on the server end as all other uses of `__experimentalSkipSerialization`  apply to both js and php sides.

We also explored adding a new `__experimentalSkipServerSerialization`, but this leaves us with yet another API that needs to be supported, just for what is currently a single edge case. 

In the end we decided that given this is a background/hidden use of the support to get around the layout quirk in the editor only, that the filter was the best approach, but open to other suggestions.

Reply:
BTW, we are seeing this as a quick fix for the most obvious/immediate issue with the wide/full layouts, done in such a way as to not serialize any changes to the block content so as to make it easier to then move to the wider/more long term fixes covering more layout flexibility mentioned on #8288

Reply:
> Because we don't want the layout supports serialized in the frontend by the server-side rendering. Without the wrapper on the frontend it adds the layout classes to the first child block instead which causes issues.

Can you clarify this, why don't we want that? There something I might be missing?

Reply:
Oh I guess because the reusable blocks have no wrapper in the frontend 🤔 I feel we should solve that first as otherwise it's going to be hacks on top of hacks. Maybe we can create the wrapper for reusable blocks v2 basically and leave the existing ones untouched.

Reply:
Adding a wrapper also solve the issue of reusable blocks showing up differently between frontend and backend.

Reply:
ok, will try and modify the approach to also add the wrapper on the frontend only when needed.

Reply:
@Mamaduka I probably won't get back to this until next week as I have some pattern category stuff to wrap up first, so feel free to take a look in the meantime if you get your other shiny things all polished up 😃

Reply:
I also need to stabilize the Table of Contents block before WP 6.4, so I'm not sure how much time I'll have outside of that. But I will give it a try if I find extra time :)